app/assets/javascripts/modules/track-responses.js

```diff
@@ -0,0 +1,93 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (global, GOVUK) {
+  'use strict'
+
+  GOVUK.Modules.TrackResponses = function () {
+    this.start = function (element) {
+      track(element[0])
+    }
+
+    function getQuestionKey(submittedForm) {
+      return submittedForm.getAttribute('data-question-key');
+    }
+
+    function getResponseLabelsForMultipleChoice(submittedForm) {
+      var labels = []
+      var checkedOptions = submittedForm.querySelectorAll('input:checked')
+
+      if (checkedOptions.length) {
+        checkedOptions.forEach(function (checkedOption) {
+          var checkedOptionId = checkedOption.getAttribute('id')
+          var checkedOptionLabel = submittedForm.querySelectorAll('label[for="' + checkedOptionId + '"]')
+          
+          var eventLabel = checkedOptionLabel.length
+            ? checkedOptionLabel[0].innerText.trim()
+            : checkedOption.value
+
+          labels.push(eventLabel)
+        })
+      } else {
+        labels.push("no response")
+      }
+
+      return labels
+    }
+
+    function getResponseLabelsForSelectOption(submittedForm) {
+      var labels = []
+      var selectInputs = submittedForm.querySelectorAll('select')
+
+      if (selectInputs.length) {
+        selectInputs.forEach(function (select) {
+          var value = select.value
+
+          if (value) {
+            var label = select.options[select.selectedIndex].innerHTML
+            var eventLabel = label.length ? label : value
+            labels.push(eventLabel)
+          } else {
+            labels.push("no response")
+          }
+        })
+      }
+
+      return labels
+    }
+
+    function getResponseLabels(submittedForm) {
+      var responseLabels = [];
+      var questionType = submittedForm.getAttribute('data-type')
+      
+      switch (questionType) {
+        case 'checkbox_question':
+        case 'multiple_choice_question':
+          responseLabels = getResponseLabelsForMultipleChoice(submittedForm)
+          break;
+        
+        case 'country_select_question':
+          responseLabels = getResponseLabelsForSelectOption(submittedForm)
+          break;
+        
+        default:
+          break;
+      }
```

Comment:
Suggestion that can be ignored - it might be a little clear if rather than setting `var responseLabels = [];` it is just declared on line 60 (`var responseLabels`). Then on set ` responseLabels = []` on line 74 so that `[]` is obviously the default behaviour:
```javascript
default: responseLabels = []
break
```